internal/accessor/accessor.go

```diff
@@ -123,7 +123,7 @@ func (a *accessor) HTTP(addr string, prefix string) NoLeaderRunnableFunc {
 		srv := http.Server{Handler: mux, Addr: addr}
 		go func() {
 			<-ctx.Done()
-			_ = srv.Shutdown(context.TODO())
+			_ = srv.Shutdown(ctx)
```

Comment:
it'll not work since it's going to use a closed context. we need to use a new one instead